### PR TITLE
Validations specs perfomance

### PR DESCRIPTION
**Pull request by iLynette**
## Project requirements
Add the following validations:

- For the User model:

- Name must not be blank.

- PostsCounter must be an integer greater than or equal to zero.

For the Post model:

- Title must not be blank.

- Title must not exceed 250 characters.

- CommentsCounter must be an integer greater than or equal to zero.

- LikesCounter must be an integer greater than or equal to zero.

- Add unit specs for all of your models' methods and validations.

- Add flash messages in the create actions in all your controllers.

- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -17,9 +17,13 @@ def create
     user = current_user
```
Comment (medaminedev66) [inline app/controllers/posts_controller.rb:17]: - [ ] Following the project requirements, you should solve N+1 problem when fetching all posts and their comments for a user as shown in the [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=ABIFLAZ4K2ZHA777536NECDBHORQS).
For example in the posts#index action, you should solve N+1 problem for the following query: `@posts = @user.posts`.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @iLynette  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- `User` model has validations
- `Post` model has validations
- Your project has unit tests for `User`, `Post`, `Comment`, and `Like` models.
- posts#create, comments#create, and likes#create actions use flash messages for successful and not successful paths.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (iLynette) [reply app/controllers/posts_controller.rb:17]: Thank you @medaminedev66, I have fixed it.

Comment (rloterh) [reply app/controllers/posts_controller.rb:17]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Lynette,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Highlights 👍
- Models validations properly implemented
- N + 1 problem properly solved
- Good coverage for model specs

<br>
Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iLynette** [comment]
Thank you @rloterh ☺️

**Pull request merged**